packages/graph-engine/src/nodes/math/cos.ts

```
@@ -9,7 +9,10 @@ export const type = NodeTypes.COS;
  * @param state
  * @returns
  */
-export const process = (input) => Math.cos(input.value);
+export const process = (input) => {
+  if (input.input === undefined) return undefined;
```

We should probably throw an error if its undefined here instead